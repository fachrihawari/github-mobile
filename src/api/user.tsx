import axios from "axios";

import config from "../config";

export const profile = (username: string, password: string) => {
  return axios.get(`${config.apiUrl}/user`, {
    auth: {
      username,
      password
    }
  });
};
