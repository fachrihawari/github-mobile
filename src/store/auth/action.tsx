import { FETCH_USER_REQUEST, LOGOUT } from "./constant";
import { createAction } from "../helpers";

export const fetchUserRequest = (username: string, password: string, OTP: string='') =>
  createAction(FETCH_USER_REQUEST, { username, password, OTP });

export const logout = () => createAction(LOGOUT);
