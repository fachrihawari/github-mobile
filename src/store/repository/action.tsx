import { FETCH_COMMIT_REQUEST } from "./constant";
import { createAction } from "../helpers";

export const fetchCommitRequest = (
  repository: string,
  page?: number,
  perPage?: number
) => createAction(FETCH_COMMIT_REQUEST, { repository, page, perPage });
