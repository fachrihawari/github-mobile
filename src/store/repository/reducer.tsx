import { AnyAction } from "redux";
import {
  FETCH_COMMIT_BEGIN,
  FETCH_COMMIT_SUCCESS,
  FETCH_COMMIT_FAILURE
} from "./constant";
import { transformCommit } from "./transform";

export interface ICommit {
  sha: string;
  date: Date;
  message: string;
  authorName: string;
  authorEmail: string;
  authorUsername: string;
  authorAvatarUrl: string;
  authorUrl: string;
}

export interface IState {
  isLoading: boolean;
  error: string | null;
  commits: Array<ICommit>;
  page: number;
  perPage: number;
}

const initialState: IState = {
  isLoading: false,
  error: null,
  page: 1,
  perPage: 10,
  commits: []
};

export default (state: IState = initialState, action: AnyAction): IState => {
  switch (action.type) {
    case FETCH_COMMIT_BEGIN:
      return {
        ...state,
        isLoading: true,
        error: null
      };
    case FETCH_COMMIT_SUCCESS:
      return {
        ...state,
        isLoading: false,
        page: action.payload.currentPage,
        commits: action.payload.commits.map(transformCommit)
      };
    case FETCH_COMMIT_FAILURE:
      return {
        ...state,
        isLoading: false,
        error: action.payload.error
      };
    default:
      return state;
  }
};
