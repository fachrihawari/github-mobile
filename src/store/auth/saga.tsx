import { AnyAction } from "redux";
import { takeLatest, put, call, delay } from "redux-saga/effects";

import {
  FETCH_USER_REQUEST,
  FETCH_USER_BEGIN,
  FETCH_USER_SUCCESS,
  FETCH_USER_FAILURE
} from "./constant";
import * as api from "../../api/user";
import { parseError } from "../helpers";

function* userRequest({ payload }: AnyAction) {
  try {
    yield put({ type: FETCH_USER_BEGIN });
    const { data } = yield call(
      api.profile,
      payload.username,
      payload.password,
      payload.OTP
    );
    yield put({
      type: FETCH_USER_SUCCESS,
      payload: {
        profile: data
      }
    });
  } catch (error) {
    console.log(error.response)
    
    let needOTP = false
    const { headers } = error.response
    if (headers['x-github-otp']) {
      const otpType = ['sms', 'app']
      const needType = headers['x-github-otp'].replace('required;', '').trim()
      needOTP = otpType.indexOf(needType) > -1

      try {
        yield call(
          api.authorize,
          payload.username,
          payload.password
        );
      } catch (error) {}

    }

    yield put({
      type: FETCH_USER_FAILURE,
      payload: {
        error: parseError(error),
        needOTP
      }
    });
  }
}
export default [takeLatest(FETCH_USER_REQUEST, userRequest)];
