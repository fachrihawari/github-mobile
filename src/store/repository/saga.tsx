import { AnyAction } from "redux";
import { takeLatest, put, call, delay } from "redux-saga/effects";

import {
  FETCH_COMMIT_REQUEST,
  FETCH_COMMIT_BEGIN,
  FETCH_COMMIT_SUCCESS,
  FETCH_COMMIT_FAILURE
} from "./constant";
import * as api from "../../api/repository";
import { parseError } from "../helpers";

function* commitRequest({ payload }: AnyAction) {
  try {
    yield put({ type: FETCH_COMMIT_BEGIN });
    const { data } = yield call(
      api.commits,
      payload.repository,
      payload.page,
      payload.perPage
    );
    yield put({
      type: FETCH_COMMIT_SUCCESS,
      payload: {
        commits: data,
        currentPage: payload.page < 1 ? 1 : payload.page
      }
    });
  } catch (error) {
    yield put({
      type: FETCH_COMMIT_FAILURE,
      payload: {
        error: parseError(error)
      }
    });
  }
}
export default [takeLatest(FETCH_COMMIT_REQUEST, commitRequest)];
