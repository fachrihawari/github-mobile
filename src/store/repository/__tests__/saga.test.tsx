import { SagaIterator } from 'redux-saga';
import { call, put } from 'redux-saga/effects';
import * as repositoryApi from '../../../api/repository';
import { fetchCommitBegin, fetchCommitFailure, fetchCommitRequest, fetchCommitSuccess } from '../action';
import { commitRequest } from '../saga';

const commitData = {
  'sha': '825e1c087c314cb06345d6e6e521cb3739f7687b',
  'commit': {
    'author': {
      'name': 'Joshua Gross',
      'email': 'joshuagross@fb.com',
      'date': '2019-08-13T20:33:20Z'
    },
    'committer': {
      'name': 'Facebook Github Bot',
      'email': 'facebook-github-bot@users.noreply.github.com',
      'date': '2019-08-13T20:37:24Z'
    },
    'message': 'Commands: support Float arguments\n\nSummary: Support codegen\'ing commands with Float arguments.\n\nReviewed By: mdvacca\n\nDifferential Revision: D16785534\n\nfbshipit-source-id: 8174ae40762c1114b87a023cb2b69b2515dc6e23',
    'tree': {
      'sha': '6aa4f8972b798d30c92f61d454ddb9494746b198',
      'url': 'https://api.github.com/repos/facebook/react-native/git/trees/6aa4f8972b798d30c92f61d454ddb9494746b198'
    },
    'url': 'https://api.github.com/repos/facebook/react-native/git/commits/825e1c087c314cb06345d6e6e521cb3739f7687b',
    'comment_count': 0,
    'verification': {
      'verified': false,
      'reason': 'unsigned',
      'signature': null,
      'payload': null
    }
  },
  'url': 'https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b',
  'html_url': 'https://github.com/facebook/react-native/commit/825e1c087c314cb06345d6e6e521cb3739f7687b',
  'comments_url': 'https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b/comments',
  'author': {
    'login': 'JoshuaGross',
    'id': 70602,
    'node_id': 'MDQ6VXNlcjcwNjAy',
    'avatar_url': 'https://avatars1.githubusercontent.com/u/70602?v=4',
    'gravatar_id': '',
    'url': 'https://api.github.com/users/JoshuaGross',
    'html_url': 'https://github.com/JoshuaGross',
    'followers_url': 'https://api.github.com/users/JoshuaGross/followers',
    'following_url': 'https://api.github.com/users/JoshuaGross/following{/other_user}',
    'gists_url': 'https://api.github.com/users/JoshuaGross/gists{/gist_id}',
    'starred_url': 'https://api.github.com/users/JoshuaGross/starred{/owner}{/repo}',
    'subscriptions_url': 'https://api.github.com/users/JoshuaGross/subscriptions',
    'organizations_url': 'https://api.github.com/users/JoshuaGross/orgs',
    'repos_url': 'https://api.github.com/users/JoshuaGross/repos',
    'events_url': 'https://api.github.com/users/JoshuaGross/events{/privacy}',
    'received_events_url': 'https://api.github.com/users/JoshuaGross/received_events',
    'type': 'User',
    'site_admin': false
  },
  'committer': {
    'login': 'facebook-github-bot',
    'id': 6422482,
    'node_id': 'MDQ6VXNlcjY0MjI0ODI=',
    'avatar_url': 'https://avatars3.githubusercontent.com/u/6422482?v=4',
    'gravatar_id': '',
    'url': 'https://api.github.com/users/facebook-github-bot',
    'html_url': 'https://github.com/facebook-github-bot',
    'followers_url': 'https://api.github.com/users/facebook-github-bot/followers',
    'following_url': 'https://api.github.com/users/facebook-github-bot/following{/other_user}',
    'gists_url': 'https://api.github.com/users/facebook-github-bot/gists{/gist_id}',
    'starred_url': 'https://api.github.com/users/facebook-github-bot/starred{/owner}{/repo}',
    'subscriptions_url': 'https://api.github.com/users/facebook-github-bot/subscriptions',
    'organizations_url': 'https://api.github.com/users/facebook-github-bot/orgs',
    'repos_url': 'https://api.github.com/users/facebook-github-bot/repos',
    'events_url': 'https://api.github.com/users/facebook-github-bot/events{/privacy}',
    'received_events_url': 'https://api.github.com/users/facebook-github-bot/received_events',
    'type': 'User',
    'site_admin': false
  },
  'parents': [
    {
      'sha': '724fe11472cb874ce89657b2c3e7842feff04205',
      'url': 'https://api.github.com/repos/facebook/react-native/commits/724fe11472cb874ce89657b2c3e7842feff04205',
      'html_url': 'https://github.com/facebook/react-native/commit/724fe11472cb874ce89657b2c3e7842feff04205'
    }
  ]
}
const payload = {
  repository: 'Username',
  page: 0,
  perPage: 10,
}

describe('userRequest Saga', () => {
  let getUserGenerator: SagaIterator

  function setup() {
    const action = fetchCommitRequest(payload.repository, payload.page, payload.perPage)
    getUserGenerator = commitRequest(action);
  }

  beforeEach(() => {
    setup()
  });

  it('should dispatch the fetchCommitSuccess action for page < 1', () => {
    const putBegin = getUserGenerator.next().value;
    expect(putBegin).toEqual(put(fetchCommitBegin()));
    expect(putBegin).toMatchSnapshot();

    const callDescriptor = getUserGenerator.next().value;
    expect(callDescriptor).toEqual(call(repositoryApi.commits, payload.repository, payload.page, payload.perPage));
    expect(callDescriptor).toMatchSnapshot();
    
    const putFailure = getUserGenerator.next({ data: [commitData], currentPage: 0 }).value;
    expect(putFailure).toEqual(put(fetchCommitSuccess([commitData], 1)));
    expect(putFailure).toMatchSnapshot();
  });

  it('should dispatch the fetchCommitSuccess action for page >= 1', () => {
    payload.page = 1
    setup()

    const putBegin = getUserGenerator.next().value;
    expect(putBegin).toMatchSnapshot();
    expect(putBegin).toEqual(put(fetchCommitBegin()));

    const callDescriptor = getUserGenerator.next().value;
    expect(callDescriptor).toEqual(call(repositoryApi.commits, payload.repository, payload.page, payload.perPage));
    expect(callDescriptor).toMatchSnapshot();
    
    const putFailure = getUserGenerator.next({ data: [commitData], currentPage: 1 }).value;
    expect(putFailure).toEqual(put(fetchCommitSuccess([commitData], 1)));
    expect(putFailure).toMatchSnapshot();
  });

  it('should dispatch the fetchCommitFailure action', () => {
    const putBegin = getUserGenerator.next().value;
    expect(putBegin).toEqual(put(fetchCommitBegin()));
    expect(putBegin).toMatchSnapshot();

    const callDescriptor = getUserGenerator.next().value;
    expect(callDescriptor).toEqual(call(repositoryApi.commits, payload.repository, payload.page, payload.perPage));
    expect(callDescriptor).toMatchSnapshot();

    const error = {
      response: {
        data: {
          message: 'Not Found'
        },
        headers: {}
      }
    }
    // @ts-ignore
    const putFailure = getUserGenerator.throw(error).value 
    expect(putFailure).toEqual(put(fetchCommitFailure(error.response.data.message)));
    expect(putFailure).toMatchSnapshot();
  });

  
  it('should dispatch the fetchCommitFailure action if failing network', () => {
    const putBegin = getUserGenerator.next().value;
    expect(putBegin).toEqual(put(fetchCommitBegin()));
    expect(putBegin).toMatchSnapshot();

    const callDescriptor = getUserGenerator.next().value;
    expect(callDescriptor).toEqual(call(repositoryApi.commits, payload.repository, payload.page, payload.perPage));
    expect(callDescriptor).toMatchSnapshot();
    
    const putFailure = getUserGenerator.next().value;
    expect(putFailure).toEqual(put(fetchCommitFailure('You\'re offline!')));
    expect(putFailure).toMatchSnapshot();
  });
});