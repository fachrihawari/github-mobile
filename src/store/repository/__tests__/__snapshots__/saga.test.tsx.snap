// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userRequest Saga should dispatch the fetchCommitFailure action 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "repository/FETCH_COMMIT_BEGIN",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitFailure action 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "Username",
      1,
      10,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitFailure action 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "payload": Object {
        "error": "Not Found",
      },
      "type": "repository/FETCH_COMMIT_FAILURE",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitFailure action if failing network 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "repository/FETCH_COMMIT_BEGIN",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitFailure action if failing network 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "Username",
      1,
      10,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitFailure action if failing network 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "payload": Object {
        "error": "You're offline!",
      },
      "type": "repository/FETCH_COMMIT_FAILURE",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page < 1 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "repository/FETCH_COMMIT_BEGIN",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page < 1 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "Username",
      0,
      10,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page < 1 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "payload": Object {
        "commits": Array [
          Object {
            "author": Object {
              "avatar_url": "https://avatars1.githubusercontent.com/u/70602?v=4",
              "events_url": "https://api.github.com/users/JoshuaGross/events{/privacy}",
              "followers_url": "https://api.github.com/users/JoshuaGross/followers",
              "following_url": "https://api.github.com/users/JoshuaGross/following{/other_user}",
              "gists_url": "https://api.github.com/users/JoshuaGross/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/JoshuaGross",
              "id": 70602,
              "login": "JoshuaGross",
              "node_id": "MDQ6VXNlcjcwNjAy",
              "organizations_url": "https://api.github.com/users/JoshuaGross/orgs",
              "received_events_url": "https://api.github.com/users/JoshuaGross/received_events",
              "repos_url": "https://api.github.com/users/JoshuaGross/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/JoshuaGross/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/JoshuaGross/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/JoshuaGross",
            },
            "comments_url": "https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b/comments",
            "commit": Object {
              "author": Object {
                "date": "2019-08-13T20:33:20Z",
                "email": "joshuagross@fb.com",
                "name": "Joshua Gross",
              },
              "comment_count": 0,
              "committer": Object {
                "date": "2019-08-13T20:37:24Z",
                "email": "facebook-github-bot@users.noreply.github.com",
                "name": "Facebook Github Bot",
              },
              "message": "Commands: support Float arguments

Summary: Support codegen'ing commands with Float arguments.

Reviewed By: mdvacca

Differential Revision: D16785534

fbshipit-source-id: 8174ae40762c1114b87a023cb2b69b2515dc6e23",
              "tree": Object {
                "sha": "6aa4f8972b798d30c92f61d454ddb9494746b198",
                "url": "https://api.github.com/repos/facebook/react-native/git/trees/6aa4f8972b798d30c92f61d454ddb9494746b198",
              },
              "url": "https://api.github.com/repos/facebook/react-native/git/commits/825e1c087c314cb06345d6e6e521cb3739f7687b",
              "verification": Object {
                "payload": null,
                "reason": "unsigned",
                "signature": null,
                "verified": false,
              },
            },
            "committer": Object {
              "avatar_url": "https://avatars3.githubusercontent.com/u/6422482?v=4",
              "events_url": "https://api.github.com/users/facebook-github-bot/events{/privacy}",
              "followers_url": "https://api.github.com/users/facebook-github-bot/followers",
              "following_url": "https://api.github.com/users/facebook-github-bot/following{/other_user}",
              "gists_url": "https://api.github.com/users/facebook-github-bot/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/facebook-github-bot",
              "id": 6422482,
              "login": "facebook-github-bot",
              "node_id": "MDQ6VXNlcjY0MjI0ODI=",
              "organizations_url": "https://api.github.com/users/facebook-github-bot/orgs",
              "received_events_url": "https://api.github.com/users/facebook-github-bot/received_events",
              "repos_url": "https://api.github.com/users/facebook-github-bot/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/facebook-github-bot/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/facebook-github-bot/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/facebook-github-bot",
            },
            "html_url": "https://github.com/facebook/react-native/commit/825e1c087c314cb06345d6e6e521cb3739f7687b",
            "parents": Array [
              Object {
                "html_url": "https://github.com/facebook/react-native/commit/724fe11472cb874ce89657b2c3e7842feff04205",
                "sha": "724fe11472cb874ce89657b2c3e7842feff04205",
                "url": "https://api.github.com/repos/facebook/react-native/commits/724fe11472cb874ce89657b2c3e7842feff04205",
              },
            ],
            "sha": "825e1c087c314cb06345d6e6e521cb3739f7687b",
            "url": "https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b",
          },
        ],
        "currentPage": 1,
      },
      "type": "repository/FETCH_COMMIT_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page >= 1 1`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "type": "repository/FETCH_COMMIT_BEGIN",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page >= 1 2`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "args": Array [
      "Username",
      1,
      10,
    ],
    "context": null,
    "fn": [Function],
  },
  "type": "CALL",
}
`;

exports[`userRequest Saga should dispatch the fetchCommitSuccess action for page >= 1 3`] = `
Object {
  "@@redux-saga/IO": true,
  "combinator": false,
  "payload": Object {
    "action": Object {
      "payload": Object {
        "commits": Array [
          Object {
            "author": Object {
              "avatar_url": "https://avatars1.githubusercontent.com/u/70602?v=4",
              "events_url": "https://api.github.com/users/JoshuaGross/events{/privacy}",
              "followers_url": "https://api.github.com/users/JoshuaGross/followers",
              "following_url": "https://api.github.com/users/JoshuaGross/following{/other_user}",
              "gists_url": "https://api.github.com/users/JoshuaGross/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/JoshuaGross",
              "id": 70602,
              "login": "JoshuaGross",
              "node_id": "MDQ6VXNlcjcwNjAy",
              "organizations_url": "https://api.github.com/users/JoshuaGross/orgs",
              "received_events_url": "https://api.github.com/users/JoshuaGross/received_events",
              "repos_url": "https://api.github.com/users/JoshuaGross/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/JoshuaGross/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/JoshuaGross/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/JoshuaGross",
            },
            "comments_url": "https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b/comments",
            "commit": Object {
              "author": Object {
                "date": "2019-08-13T20:33:20Z",
                "email": "joshuagross@fb.com",
                "name": "Joshua Gross",
              },
              "comment_count": 0,
              "committer": Object {
                "date": "2019-08-13T20:37:24Z",
                "email": "facebook-github-bot@users.noreply.github.com",
                "name": "Facebook Github Bot",
              },
              "message": "Commands: support Float arguments

Summary: Support codegen'ing commands with Float arguments.

Reviewed By: mdvacca

Differential Revision: D16785534

fbshipit-source-id: 8174ae40762c1114b87a023cb2b69b2515dc6e23",
              "tree": Object {
                "sha": "6aa4f8972b798d30c92f61d454ddb9494746b198",
                "url": "https://api.github.com/repos/facebook/react-native/git/trees/6aa4f8972b798d30c92f61d454ddb9494746b198",
              },
              "url": "https://api.github.com/repos/facebook/react-native/git/commits/825e1c087c314cb06345d6e6e521cb3739f7687b",
              "verification": Object {
                "payload": null,
                "reason": "unsigned",
                "signature": null,
                "verified": false,
              },
            },
            "committer": Object {
              "avatar_url": "https://avatars3.githubusercontent.com/u/6422482?v=4",
              "events_url": "https://api.github.com/users/facebook-github-bot/events{/privacy}",
              "followers_url": "https://api.github.com/users/facebook-github-bot/followers",
              "following_url": "https://api.github.com/users/facebook-github-bot/following{/other_user}",
              "gists_url": "https://api.github.com/users/facebook-github-bot/gists{/gist_id}",
              "gravatar_id": "",
              "html_url": "https://github.com/facebook-github-bot",
              "id": 6422482,
              "login": "facebook-github-bot",
              "node_id": "MDQ6VXNlcjY0MjI0ODI=",
              "organizations_url": "https://api.github.com/users/facebook-github-bot/orgs",
              "received_events_url": "https://api.github.com/users/facebook-github-bot/received_events",
              "repos_url": "https://api.github.com/users/facebook-github-bot/repos",
              "site_admin": false,
              "starred_url": "https://api.github.com/users/facebook-github-bot/starred{/owner}{/repo}",
              "subscriptions_url": "https://api.github.com/users/facebook-github-bot/subscriptions",
              "type": "User",
              "url": "https://api.github.com/users/facebook-github-bot",
            },
            "html_url": "https://github.com/facebook/react-native/commit/825e1c087c314cb06345d6e6e521cb3739f7687b",
            "parents": Array [
              Object {
                "html_url": "https://github.com/facebook/react-native/commit/724fe11472cb874ce89657b2c3e7842feff04205",
                "sha": "724fe11472cb874ce89657b2c3e7842feff04205",
                "url": "https://api.github.com/repos/facebook/react-native/commits/724fe11472cb874ce89657b2c3e7842feff04205",
              },
            ],
            "sha": "825e1c087c314cb06345d6e6e521cb3739f7687b",
            "url": "https://api.github.com/repos/facebook/react-native/commits/825e1c087c314cb06345d6e6e521cb3739f7687b",
          },
        ],
        "currentPage": 1,
      },
      "type": "repository/FETCH_COMMIT_SUCCESS",
    },
    "channel": undefined,
  },
  "type": "PUT",
}
`;
